app/src/main/java/kr/co/yogiyo/rookiephotoapp/diary/db/LocalDiaryManager.java

```diff
@@ -0,0 +1,120 @@
+package kr.co.yogiyo.rookiephotoapp.diary.db;
+
+import android.arch.persistence.room.Room;
+import android.content.Context;
+
+import java.util.Date;
+
+import io.reactivex.Completable;
+import io.reactivex.CompletableObserver;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.functions.Action;
+import io.reactivex.functions.Consumer;
+import io.reactivex.schedulers.Schedulers;
+
+public class LocalDiaryManager {
+    private static final String DB_NAME = "Diaries.db";
+    private Context context;
+    private static LocalDiaryManager _instance;
+    private DiaryDatabase db;
+
+    public static LocalDiaryManager getInstance(Context context) {
+        if (_instance == null) {
+            _instance = new LocalDiaryManager(context);
+        }
+        return _instance;
+    }
+
+    public LocalDiaryManager(Context context) {
+        this.context = context;
+        db = Room.databaseBuilder(context, DiaryDatabase.class, DB_NAME).build();
+    }
+
+    public void insertDiary(final DiaryDatabaseCallback databaseCallback, final int idx, final Date date, final String image, final String description) {
+        Completable.fromAction(new Action() {
+            @Override
+            public void run() {
+                Diary diary = new Diary(idx,date, image, description);
+                db.diaryDao().insertDiary(diary);
+            }
+        }).observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io()).subscribe(new CompletableObserver() {
+            @Override
+            public void onSubscribe(Disposable d) {
+            }
+
+            @Override
+            public void onComplete() {
+                databaseCallback.onDiaryAdded();
+            }
+
+            @Override
+            public void onError(Throwable e) {
+            }
+        });
+    }
+
+    public void findDiaryById(final DiaryDatabaseCallback databaseCallback, final String diaryId) {
+        db.diaryDao().findDiaryById(diaryId)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(new Consumer<Diary>() {
+                    @Override
+                    public void accept(Diary diary) {
+                        databaseCallback.onDiaryByIdFinded(diary);
+                    }
+                }).isDisposed();
+    }
+
+    public void updateDiary(final DiaryDatabaseCallback databaseCallback, final Diary diary, final Date date, final String image, final String description) {
+
+        diary.setDate(date);
+        diary.setImage(image);
+        diary.setDescription(description);
+
+        Completable.fromAction(new Action() {
+            @Override
+            public void run() {
+                db.diaryDao().updateDiary(diary);
+            }
+        }).subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).subscribe(new CompletableObserver() {
```

Comment:
```suggestion
        }).subscribeOn(Schedulers.io())
        .observeOn(AndroidSchedulers.mainThread())
        .subscribe(new CompletableObserver() {
```

이렇게 한줄로 쓰면 알아보기 힘드니까 띄어쓰기 적용해주세요.

Reply:
네 적용했습니다